src/pytest_bdd/scenario.py

```diff
@@ -249,9 +249,16 @@ def collect_example_parametrizations(
     contexts = list(templated_scenario.examples.as_contexts())
     if not contexts:
         return None
+    return [pytest.param(context[0], id="-".join(context[0].values()), marks=_register_marks(context[1])) for context in contexts]
 
-    return [pytest.param(context, id="-".join(context.values())) for context in contexts]
 
+def _register_marks(marks_object) -> list[pytest.mark]:
+    marks_list = []
+    for marks in marks_object:
+        mark = getattr(pytest.mark, marks)
+        marks_list.append(mark)
+
+    return marks_list
```

Comment:
```suggestion
    return [getattr(pytest.mark, marks) for marks in marks_object]
```

Reply:
Couple of quick questions:

1. I assume that "marks" are the pytest concept that `pytest-bdd` is using to implement Cucumber/Gherkin "tags"?
2. If I'm understanding the code above correctly, it seems like the thing being passed to this method is actually the `set[str]` of tags. Did I get that right?

If so, then I would suggest that instead this method should look something like:
```
def _register_tags_as_marks(tags) -> list[pytest.mark]:
    return [getattr(pytest.mark, tag) for tag in tags]
```

Reply:
to answer your questions:
1. Yep! pytest-bdd/Gherkin tags become pytest marks
2. Correct

Reply:
From what I understood:
tags = tags at a higher level
marks = tags at the lower levels (like on examples)